### PR TITLE
Miscellaneous related RZ and thermal modifications to PorousFlow and TensorMechanics

### LINKED ISSUE #10152: Misc RZ and Thermal enhancements to PorousFlow and TensorMechanics
## Rationale and Description

A number of small and connected changes regarding RZ coordinate systems and thermal coupling in PorousFlow and TensorMechanics:

1.   PorousFlowEffectiveStressCoupling is incorrect for RZ coordinate systems.  It needs to be modified along the lines of StressDivergenceRZTensors.   Probably a seperate class isn't necessary since this is such a simple Kernel anyway - just a RZ-check will do.  All that's required is to replace _grad_test with _grad_test + _test/r  when component=0 and the RZ-check is true

2. The OffDiag Jacobian is currently zero for thermally-coupled StressDivergenceRZTensors.C.  It needs to be nonzero.  I will fix it **for small strains only** as that's what i'm interested in

3. The PorousFlow Actions only work for XYZ coordinate systems and i want RZ as well.  I need to check the coord system, and use StressDivergenceRZTensors if appropriate

4.  The PorousFlow Actions do not add the thermal_eigenstrain_name to the StressDivergenceTensors at the moment, and i need to add that

## Impact

Allow RZ coordinate systems in mechanically-coupled PorousFlow simulations.
Allow more accurate calculation of Jacobian terms in thermally-coupled simulations

### PR DESCRIPTION
**Pull request by WilkAndy**
PorousFlowActionBase now records the Problem coordinate system and adds the correct StressDivergence-type actions for XYZ and RZ coordinate systems.  It also allows users to specify their thermal_eigenstrain_name in order that derivatives wrt temperature are correctly inserted into the Jacobian (for small strain, at least)

PorousFlowEffectiveStressCoupling now produces residuals and Jacobians that are correct in XYZ and RZ coordinates systems

StressDivergenceRZTensors now correctly calculates OffDiag Jacobian contributions of derivatives wrt the temperature (for small strain)

Fixes #10152

### CODE REVIEW

**Comment by permcody** [comment]
@bwspenc or @dschwen - Could I get one of you to take a look at this PR? With all the changes to the way the coordinate system has changed in TM lately, I want to make sure there's nothing in here that is going in a different direction. Thanks

**Comment by WilkAndy** [comment]
It's not tonnes of code.   Perhaps @cpgr could look at the PorousFlow stuff, which is most of this PR.

**Comment by WilkAndy** [comment]
Thanks @cpgr ,   a

File: modules/porous_flow/src/actions/PorousFlowActionBase.C [hunk 1]
```diff
@@ -66,6 +74,15 @@ PorousFlowActionBase::PorousFlowActionBase(const InputParameters & params)
 void
 PorousFlowActionBase::act()
 {
+  const auto & all_subdomains = _problem->mesh().meshSubdomains();
+  if (all_subdomains.empty())
+    mooseError("No subdomains found");
+  _coord_system = _problem->getCoordSystem(*all_subdomains.begin());
+  for (auto subdomain : all_subdomains)
```
Comment (permcody) [inline modules/porous_flow/src/actions/PorousFlowActionBase.C]: You should probably get in the habit of using an `auto &` in range-for loops. You don't need it for something like this where the type is a POD, but it matters when you are looping over a more complex type. It's just good to be consistent.

File: modules/porous_flow/test/tests/jacobian/eff_stress03.i [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    max = 0
+  [../]
+  [./ppgas]
+    type = RandomIC
+    variable = ppgas
+    min = 0
+    max = 1
+  [../]
+  [./massfrac_ph0_sp0]
+    type = RandomIC
```
Comment (permcody) [inline modules/porous_flow/test/tests/jacobian/eff_stress03.i]: If you are going to use RandomIC, you should restrict this test to run on one proc/one thread. We are eventually going to run the test suite in parallel in the modules (or we'd like to and it'll take a long time to clean these up). RandomIC doesn't generate the same random field in parallel. That's going to take more work.

File: modules/porous_flow/test/tests/jacobian/tests [hunk 3]
```diff
@@ -162,6 +162,18 @@
     ratio_tol = 1E-7
     difference_tol = 1E10
   [../]
+  [./eff_stress03]
+    type = 'PetscJacobianTester'
+    input = 'eff_stress03.i'
+    ratio_tol = 1E-7
+    difference_tol = 1E10
```
Comment (permcody) [inline modules/porous_flow/test/tests/jacobian/tests]: Start restricting these tests if you know they won't work in parallel or with threads: `max_parallel = 1, max_threads=1`

**Review by permcody — CHANGES_REQUESTED** [review]

File: modules/porous_flow/test/tests/jacobian/eff_stress03.i (hunk 2 above)
Comment (WilkAndy) [reply modules/porous_flow/test/tests/jacobian/eff_stress03.i]: Aha, that's good to know about RandomIC and parallel runs.  However, in this case, does it matter?  This is just a PetscJacobian test - so any random numbers should produce a "zero" Jacobian and i shouldn't need the same random numbers each time.  Right?

Comment (permcody) [reply modules/porous_flow/test/tests/jacobian/eff_stress03.i]: Good question, you're right. It probably doesn't matter in this case.

Comment (WilkAndy) [reply modules/porous_flow/test/tests/jacobian/eff_stress03.i]: Cool, thanks.  I found some other PorousFlow tests where we should restrict to serial, so ill include the changes in this PR.

File: modules/porous_flow/src/actions/PorousFlowActionBase.C (hunk 1 above)
Comment (WilkAndy) [reply modules/porous_flow/src/actions/PorousFlowActionBase.C]: fixed

**Review by permcody — APPROVED** [review]

**Comment by WilkAndy** [comment]
Thanks for the review and merge (another one!) @permcody ,

a

**Pull request merged**